damnit/gui/main_window.py

```diff
@@ -260,17 +267,49 @@ def autoconfigure(self, path: Path):
         self.show_default_status_message()
         self.context_dir_changed.emit(str(path))
         self.launch_update_computed_vars()
+        self.start_watching_context_file()
 
     def launch_update_computed_vars(self):
+        # Triggered when we open a proposal & when saving the context file
         log.debug("Launching subprocess to read variables from context file")
+        self.updating_vars = True
         proc = QtCore.QProcess(parent=self)
         # Show stdout & stderr with the parent process
         proc.setProcessChannelMode(QtCore.QProcess.ProcessChannelMode.ForwardedChannels)
+        proc.finished.connect(self.done_update_computed_vars)
         proc.finished.connect(proc.deleteLater)
         proc.setWorkingDirectory(str(self.context_dir))
         proc.start(sys.executable, ['-m', 'damnit.cli', 'read-context'])
         proc.closeWriteChannel()
 
+    def done_update_computed_vars(self):
+        self.updating_vars = False
+        self.vars_ctx_size_mtime = self.get_context_size_mtime()
+
+    def get_context_size_mtime(self):
+        st = self._context_path.stat()
+        return st.st_size, st.st_mtime
+
+    def poll_context_file(self):
+        size_mtime = self.get_context_size_mtime()
+        if (self.vars_ctx_size_mtime != size_mtime) and not self.updating_vars:
+            log.info("Context file changed, updating computed variables")
+            self.launch_update_computed_vars()
```

Comment:
Thanks, that's a good point. It's really fiddly to get this completely 'right', but as you say, it's better to process a version twice because we used an older stat result than miss an update by using a newer one.

Reply:
For the editor, I've made it stat the file before & after reading - if the size & mtime don't change, we can assume that they correspond with the data we've read (so long as no-one is manually setting the mtime back - but we already assume that).

Reply:
Looks good!